src/tools/wasm-as.cpp

```diff
@@ -86,13 +94,22 @@ int main(int argc, const char *argv[]) {
   if (options.debug) std::cerr << "binarification..." << std::endl;
   BufferWithRandomAccess buffer(options.debug);
   WasmBinaryWriter writer(&wasm, buffer, options.debug);
-  writer.setDebugInfo(debugInfo);
+  writer.setNamesSection(debugInfo);
```

Comment:
this looks weird, it used to be the same name, but now it is different. perhaps add a comment, "if debug info is used, then we want to emit the names section" (is that correct?)